week-06-202402/02/Juego.cpp

```diff
@@ -0,0 +1,99 @@
+#include "Juego.h"
+#include "chrono"
+
+Juego::Juego() {
+	this->gato = new Gato(25, 25);
+	cantidad_ratones = rand() % 9 + 7;
+	for (int i = 0; i < cantidad_ratones; i++) {
+		listaRatones.push_back(new Raton());
+	}
+	tiempoAnterior = chrono::steady_clock::now();
+}
+
+Juego::~Juego() {
+	for (int i = 0; i < cantidad_ratones; i++) {
+		delete listaRatones[i];
+	}
+	listaRatones.clear();
+	delete gato;
+}
+
+bool Juego::detectarColision() {
+	for (int i = 0; i < listaRatones.size(); i++) {
+		int ratonX = listaRatones[i]->getX();
+		int ratonY = listaRatones[i]->getY();
+
+		int ratonAncho = 8;
```

Comment:
As a recommendation, deleting should not be a pure abstract class. Instead, we could define it considering saving the dimensions of the objects and deleting in the area.